### PR TITLE
Added new validation rules to validate the lengths of string column types

### PR DESCRIPTION
**Pull request by yulric**
Changes should be self explanatory. Some notes:

*  Added new odm how to, retrieving the data type for a column.
* Can you confirm what ODM types can be considered to be strings?

### CODE REVIEW

File: metadata/validation-rules-list.csv [hunk 1]
```diff
@@ -5,3 +5,5 @@ duplicate_entries_found,Validates that a primary key is not invalidly duplicated
...
 missing_values_found,Validates that a mandatory column does not have null values,The ODM dictionary provides a list of parts that are represent missing. This rule ensures that mandatory columns do not contain these values.,error,Mandatory column <column_name> in table <table_name> contains missing value <invalid_value> in row index <row_index>,,v1.0.0,,all,,
+less_than_min_length,Validates the minimum length of a string type,The minLength column in the ODM dictionary documents the minimum that a part should have. This validation implements it.,error,Value <invalid_value> in row <row_index> in column <column_name> in table <table_name> has length <invalid_length> which is less than the min length of <min_length>,active,v1.0.0,,all,,
+greater_than_min_length,Validates the maximum length of a string type,The maxLength column in the ODM dictionary documents the minimum that a part should have. This validation implements it.,error,Value <invalid_value> in row <row_index> in column <column_name> in table <table_name> has length <invalid_length> which is greater than the max length of <max_length>,active,v1.0.0,,all,,
```
Comment (DougManuel) [inline metadata/validation-rules-list.csv:9]: typo? 'greater_than_min_length' --> 'greater_than_max_length'

**Comment by DougManuel** [comment]
@mathew-thomson can you clarify? I was expecting that only parts with varChar should have a minLength and maxLength. I see parts such as `a1918v` that have a min and max length and also minValue and maxValue of `see units`. 

Should there be only one or the other? Not both a minLength/maxLength AND minValue and maxValue. For `a1918v` shouldn't there only be `see units`?

**Comment by DougManuel** [comment]
> @mathew-thomson can you clarify? I was expecting that only parts with varChar should have a minLength and maxLength. I see parts such as `a1918v` that have a min and max length and also minValue and maxValue of `see units`.
> 
> Should there be only one or the other? Not both a minLength/maxLength AND minValue and maxValue. For `a1918v` shouldn't there only be `see units`?

OK. there should be only minLength/maxLength OR minValue and maxValue. We'll fix that for next RC (tomorrow or next week).

**Review by DougManuel — APPROVED** [review]

**Comment by mathew-thomson** [comment]
> > @mathew-thomson can you clarify? I was expecting that only parts with varChar should have a minLength and maxLength. I see parts such as `a1918v` that have a min and max length and also minValue and maxValue of `see units`.
> > Should there be only one or the other? Not both a minLength/maxLength AND minValue and maxValue. For `a1918v` shouldn't there only be `see units`?
> 
> OK. there should be only minLength/maxLength OR minValue and maxValue. We'll fix that for next RC (tomorrow or next week).

This is now resolved, sorry for the wait, y'all! Let me know if you find any issues still in these columns.

**Pull request merged**